app/controllers/services_controller.rb

```diff
@@ -54,6 +54,24 @@ def show
     render json: ServicesWithResourcePresenter.present(service)
   end
 
+  def init_pdf_crowd_client
+    Pdfcrowd::HtmlToPdfClient.new(Rails.configuration.x.pdfcrowd.username, Rails.configuration.x.pdfcrowd.api_key)
+  end
+
+  def html_to_pdf
+    if Rails.configuration.x.pdfcrowd.api_key && Rails.configuration.x.pdfcrowd.username
+      client = init_pdf_crowd_client
+      pdf = client.convertString(params[:html])
+      send_data pdf,
+                { type: "application/pdf",
+                  disposition: "attachment; filename*=UTF-8''#{ERB::Util.url_encode('result.pdf')} }" }
+    else
+      render(status: 500)
+    end
+  rescue Pdfcrowd::Error => e
+    render plain: e.getMessage, status: e.getCode
```

Comment:
Will this show up to end users? Maybe we should render a generic error message but log the more detailed error so that it shows up in Sentry.

Reply:
Sure, I've updated to return a generic errors msg. I just added a `puts` for logging. I couldn't find any other examples of logging across our codebase other than `puts`

Reply:
Hmm, I guess we're not really following best practices, but what I had in mind was using the [Rails logger](https://guides.rubyonrails.org/debugging_rails_applications.html#sending-messages).